ibm/service/power/resource_ibm_pi_host_group.go

```diff
@@ -0,0 +1,377 @@
+// Copyright IBM Corp. 2024 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package power
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM/go-sdk-core/v5/core"
+)
+
+func ResourceIBMPIHostGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMPIHostGroupCreate,
+		ReadContext:   resourceIBMPIHostGroupRead,
+		DeleteContext: resourceIBMPIHostGroupDelete,
+		UpdateContext: resourceIBMPIHostGroupUpdate,
+		Importer:      &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
+				ForceNew:     true,
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_Hosts: {
+				Description: "List of hosts to add to the group.",
+				ForceNew:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_DisplayName: {
+							Description:  "Name of the host chosen by the user.",
+							Required:     true,
+							Type:         schema.TypeString,
+							ValidateFunc: validation.NoZeroValues,
+						},
+						Attr_SysType: {
+							Description:  "System type.",
+							Required:     true,
+							Type:         schema.TypeString,
+							ValidateFunc: validation.NoZeroValues,
+						},
+					},
+				},
+				Required: true,
+				Type:     schema.TypeList,
```

Comment:
Can we change type to TypeSet . If its TypeList when we add multiple hosts the plan will always shows the diff since list is unordered .

Reply:
Same comment holds for all schema of TypeList with nested structure if the argument is either required/optional

Reply:
Updated the object to a set.